### ISSUE TITLE
ruge_stueben_amg.hpp does not compile with hipcc (clang-14)

### ISSUE THREAD

**Issue by doctorcolinsmith** [ROCmSoftwarePlatform/rocALUTION#151]
The following was reported by @dipietrantonio in https://github.com/ROCmSoftwarePlatform/rocALUTION/issues/144.  Breaking it out to a separate issue here.

rocALUTION/src/solvers/multigrid/ruge_stueben_amg.hpp:

```
#if defined(WIN32) || defined(_WIN32) || defined(__WIN32)
#else
        [[deprecated("This function will be removed in a future release. Use "
                     "SetStrengthThreshold() instead")]]
#endif
```

Had to remove the code in the else block because it wouldnt compile using hipcc (clang-14).

**Comment by cgmb** [comment]
FWIW, rocSOLVER started using this deprecation style 13 months ago and there have been no complaints thus far:
https://github.com/ROCmSoftwarePlatform/rocSOLVER/blob/rocm-5.2.3/library/include/rocsolver-aliases.h

**Comment by jsandham** [comment]
I believe this is now fixed and I think was caused by The ROCALUTION_EXPORT being located above the deprecation attribute. This is now fixed in master branch. See https://github.com/ROCmSoftwarePlatform/rocALUTION/commit/f31b90bea1a4c9232c64211fca2d0b4678f25fc7

**Comment by jsandham** [comment]
Closing as fixed for rocm-5.3.
